### ISSUE TITLE
sp-kill-hybrid-sexp doesn't always remove empty lines

### ISSUE THREAD

**Issue by edam** [Fuco1/smartparens#974]
`sp-kill-hybrid-sexp` doesn't always remove empty lines if there are only empty lines between point at the close paren.

E.g.: (point is `|`)
```javascript
foo() {
    |

}
```
In this state, `sp-kill-hybrid-sexp` has no effect at all.  Whereas it should produce:
```javascript
foo() {
    |
}
```
To illustrate this point, if we started with:
```javascript
foo() {
    |

    something();
}
```
then `sp-kill-hybrid-sexp` works as expected and produces:
```javascript
foo() {
    |
    something();
}
```
Somehow, the lack of `something();` before the close paren prevents `sp-kill-hybrid-sexp` from working!

## Environment & version information

- `smartparens` version: 1.11.0
- Active major-mode: c-mode (or js-mode)
- Emacs version: GNU Emacs 26.1 (build 2, x86_64pc-linux-gnu, GTK+ Version 3.24.4) of 2019-02-03, modified by Debian
- OS: Debian "buster" x86_64 GNU/Linux

**Comment by edam** [comment]
The underlying problem appears to be that in this case:
```javascript
foo() {
    |

}
```
`sp-get-hybrid-sexp` returns `:end` 13, which is where point is (at the end of the line). But in this case:
```javascript
foo() {
    |

    x
}
```
`sp-get-hybrid-sexp` returns `:end` 9, which is at the beginning of the line that point is on.

Which is the correct behaviour? The comment for `sp-get-hybrid-sexp` says it returns

> the smallest balanced region containing the point while not expanding further than the current line.

So does that mean that the first case is incorrect, since a region with `:end` 9 does not contain point?

**Comment by Fuco1** [comment]
Yes that seems not to be correct.  I think there might be some special behaviour in case the containing expression is completely empty.
